### PR TITLE
IB: add support for `instantiateViewController(identifier:creator:)`

### PR DESCRIPTION
**Pull request by matsune**
[`instantiateViewController(identifier:creator:)`](https://developer.apple.com/documentation/uikit/uistoryboard/3213989-instantiateviewcontroller#declaration) is now avaliable to create a view controller since iOS 13 (and OSX also has a similar method).
This allows us to inject dependencies into the view controller using storyboard, and I think SwiftGen default templates could support this.
If you intend to merge this change, I'll add an entry to changelog.

### CODE REVIEW

**Comment by SwiftGen-Eve** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

Hey 👋 I'm Eve, the friendly bot watching over SwiftGen 🤖

Thanks a lot for your contribution!



---

Seems like everything is in order 👍 You did a good job here! 🤝

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: templates/ib/scenes-swift4.stencil [hunk 1]
```diff
@@ -81,6 +81,21 @@ import {{module}}
     }
     return controller
   }
+
+  @available({% if isAppKit %}OSX 10.15{% else %}iOS 13.0, tvOS 13.0{% endif %}, *)
+  {{accessModifier}} func instantiate(creator block: @escaping (NSCoder) -> T?) -> T {
+    let identifier = {% if isAppKit %}NSStoryboard.SceneIdentifier({% endif %}self.identifier{% if isAppKit %}){% endif %}
+    {% if isAppKit %}
+    guard let controller = storyboard.storyboard
+      .__instantiate{{controller}}(withIdentifier: identifier, creator: block) as? T else {
```
Comment (djbe) [inline templates/ib/scenes-swift4.stencil]: Why the `__` for AppKit? From what I can see in the docs, the method has no `__`.

Comment (djbe) [inline templates/ib/scenes-swift4.stencil]: I don't think a `guard` with `as` cast is needed? The new `instantiate` methods are generic, and Swift should be able to infer the type to be `T`. So I think the code should just be what you have for non-appkit platforms, namely:

```swift
return storyboard.storyboard.instantiate{{controller}}(identifier: identifier, creator: block)
```

File: templates/ib/scenes-swift4.stencil [hunk 2]
```diff
@@ -92,6 +107,15 @@ import {{module}}
     }
     return controller
   }
+
+  @available({% if isAppKit %}OSX 10.15{% else %}iOS 13.0, tvOS 13.0{% endif %}, *)
+  {{accessModifier}} func instantiate(creator block: @escaping (NSCoder) -> T?) -> T {
+    guard let controller = storyboard.storyboard
```
Comment (djbe) [inline templates/ib/scenes-swift4.stencil]: Same as above, I don't think a `guard` is needed? (and same comment about `__`)

**Review by djbe — CHANGES_REQUESTED** [review]

**Comment by djbe** [comment]
@matsune I've taken the liberty of finishing up this PR. It should work, but you can take a look at it if you want.
I'll probably merge this in a day or 2 if there's no more feedback.

File: CHANGELOG.md [hunk 3]
```diff
@@ -20,6 +20,10 @@ _None_
   [@kevinstier](https://github.com/kevinstier)
   [#562](https://github.com/SwiftGen/SwiftGen/issues/562)
   [#768](https://github.com/SwiftGen/SwiftGen/pull/768)
+* IB: Added support for instantiating scenes while providing a `creator` block, commonly used for dependency injection. This feature is available starting from iOS 13, tvOS 13 and macOS 10.15.
```
Comment (SwiftGen-Eve) [inline CHANGELOG.md]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/SwiftGen/SwiftGen/blob/9998c51351f6384bd7524010e61558cb9cf44506/CHANGELOG.md#L23"></span>Line describing your entry should end with a period and 2 spaces.</td>
    </tr>
  </tbody>
</table>

**Review by djbe — APPROVED** [review]

**Comment by matsune** [comment]
@djbe Thank you for your review & fixing my PR!
I have no problem to merge this.

**Pull request merged**